packages/date-picker/src/FilterDateRangePicker/components/Trigger/TriggerButtonBase/TriggerButtonBase.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ButtonHTMLAttributes, forwardRef } from "react"
+import classNames from "classnames"
+import { OverrideClassName } from "@kaizen/component-base"
+import { Icon } from "@kaizen/component-library"
+import chevronDown from "@kaizen/component-library/icons/chevron-down.icon.svg"
+import chevronUp from "@kaizen/component-library/icons/chevron-up.icon.svg"
+import styles from "./TriggerButtonBase.module.scss"
+
+export interface TriggerButtonBaseProps
+  extends OverrideClassName<ButtonHTMLAttributes<HTMLButtonElement>> {
+  children: React.ReactNode
+  isOpen?: boolean
+}
+
+export const TriggerButtonBase = forwardRef<
+  HTMLButtonElement,
+  TriggerButtonBaseProps
+>(({ children, classNameOverride, isOpen = false, ...restProps }, ref) => (
+  <button
+    ref={ref}
+    className={classNames(styles.button, classNameOverride)}
+    aria-haspopup="true"
+    aria-expanded={isOpen}
+    {...restProps}
+  >
+    {children}
+    <Icon
+      icon={isOpen ? chevronUp : chevronDown}
+      role="presentation"
+      classNameOverride={styles.icon}
```

Comment:
Lol using our own `classNameOverride`, could we use a flex on button instead?

Reply:
Nah, we still need some consistent gap in it. Nothing wrong with using our own override though :) That's how styling and extension works :D

Reply:
Adding `gap` would achieve that too, right?

> Nothing wrong with using our own override though :) That's how styling and extension works :D
Sure, but I'd argue that we shouldn't need to since most of the use cases for it can be solved with CSS on containers which is much more scalable.

Reply:
Shh, stop being smart :P